packages/app/src/app/overmind/namespaces/dashboard/actions.ts

```diff
@@ -1026,14 +1167,51 @@ export const changeSandboxPrivacy: AsyncAction<{
   privacy: 0 | 1 | 2;
   oldPrivacy: 0 | 1 | 2;
 }> = async ({ actions, effects, state }, { id, privacy, oldPrivacy }) => {
+  const repoChangePrivacy = (p: 0 | 1 | 2) => {
+    if (
+      !state.dashboard.sandboxes.REPOS ||
+      !state.dashboard.sandboxes.REPOS[repoName]
+    ) {
+      return;
+    }
+
+    const repoSandboxes = state.dashboard.sandboxes.REPOS[repoName];
+    state.dashboard.sandboxes.REPOS = {
+      ...state.dashboard.sandboxes.REPOS,
+      [repoName]: {
+        ...repoSandboxes,
+        sandboxes: repoSandboxes?.sandboxes.map(sandbox => {
+          if (sandbox.id === id) {
+            return {
+              ...sandbox,
+              privacy: p,
+            };
+          }
+
+          return sandbox;
+        }),
+      },
+    };
+  };
   // optimistic update
-  actions.dashboard.changeSandboxPrivacyInState({ id, privacy });
+  if (isRepoPage) {
+    repoChangePrivacy(privacy);
+  } else {
+    actions.dashboard.changeSandboxPrivacyInState({ id, privacy });
```

Comment:
I would also put the repo privacy change in this function instead of 2 entry points.

Reply:
done